packages/Search/src/DTK_DistributedSearchTree_decl.hpp

```diff
@@ -111,9 +114,8 @@ void DistributedSearchTree<DeviceType>::query(
     Kokkos::View<int *, DeviceType> &ranks ) const
 {
     using Tag = typename Query::Tag;
-    DistributedSearchTreeImpl<DeviceType>::queryDispatch(
-        _comm, _distributed_tree, _local_tree, queries, indices, offset, ranks,
-        Tag{} );
+    Details::DistributedSearchTreeImpl<DeviceType>::queryDispatch(
+        *this, queries, indices, offset, ranks, Tag{} );
```

Comment:
If you need to make `DistributedSearchTreeImpl `a friend and pass `*this`, why not move `queryDispatch` to the class. Because this is basically what you are doing but in a complicated way.